### ISSUE TITLE
Test Set

### ISSUE THREAD

**Issue by sdws259** [songmzhang/DSKD#25]
Dear Authors,
     Thank you for your excellent work. However, I have a question regarding the test set in your experiments. I directly downloaded the dataset provided by MiniLLM and found that the valid.jsonl and valid.txt files in the 11_ directory of the UNIST dataset contain 23,916 samples. However, your paper mentions that this dataset has only 10,000 samples. Does it requires special handling? I noticed that the descriptions of these datasets vary across different papers. Would you be willing to share your Train/Valid/Test dataset with me? I look forward to your reply.
     Best regards.

**Comment by songmzhang** [comment]
Thanks for your attention. For evaluation, we only use 10000 of 23916 samples in the 11_ directory of the UNIST dataset, and this is also the default setting in MiniLLM. It is controlled in the following line (see the "10000" in the last parameter): https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28

**Comment by sdws259** [comment]
Ok, I have got it. Thanks~~~

**Comment by sdws259** [comment]
Dear Authors,
   I'm sorry for disturbing you again. I have met the following issue when looking at the training log file of gpt2-xl. Is this normal? The log file during your chat also exists this problem. Any ways to address it? Note that this problem also exists when evaluating the trained model on different datasets, such as unist, dolly. 

![Image](https://github.com/user-attachments/assets/8955686a-10cb-4c18-aba1-0de6645cd733)

**Comment by songmzhang** [comment]
> Dear Authors, I'm sorry for disturbing you again. I have met the following issue when looking at the training log file of gpt2-xl. Is this normal? The log file during your chat also exists this problem. Any ways to address it? Note that this problem also exists when evaluating the trained model on different datasets, such as unist, dolly.
> 
> ![Image](https://github.com/user-attachments/assets/8955686a-10cb-4c18-aba1-0de6645cd733)

Hi, I'm not sure what happened in this figure. Is this the log file during training? Why are there 23916 items of data?

**Comment by sdws259** [comment]
> > Dear Authors, I'm sorry for disturbing you again. I have met the following issue when looking at the training log file of gpt2-xl. Is this normal? The log file during your chat also exists this problem. Any ways to address it? Note that this problem also exists when evaluating the trained model on different datasets, such as unist, dolly.
> > ![Image](https://github.com/user-attachments/assets/8955686a-10cb-4c18-aba1-0de6645cd733)
> 
> Hi, I'm not sure what happened in this figure. Is this the log file during training? Why are there 23916 items of data?

我很抱歉带来困惑，这是评估训练好的模型在unist数据集上的测试日志。我想说的是在训练和测试日志中都观察到上面的“token indices sequence length is longer than the specified maximum sequence length for this model”警告，事实上这个警告看起来比较常见，好像tokenizer会自动处理，但我想知道你们有没有做过一些额外的特殊处理？

**Comment by songmzhang** [comment]
Yes, we did length truncation after tokenization on both prompts and responses after loading the dataset, so this warning doesn't matter. BTW, it seems that you use all the data in 11_ of the UNIST dataset for evaluation. However, we only used the first 10,000 samples in this directory for evaluation, as shown in the script. https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28

**Comment by sdws259** [comment]
> Yes, we did length truncation after tokenization on both prompts and responses after loading the dataset, so this warning doesn't matter. BTW, it seems that you use all the data in 11_ of the UNIST dataset for evaluation. However, we only used the first 10,000 samples in this directory for evaluation, as shown in the script.
> 
> [DSKD/scripts/eval/run_eval.sh](https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28)
> 
> Line 28 in [57d2290](/songmzhang/DSKD/commit/57d2290ce2d448be6293d97a82c6608addf33cdb)
> 
>  bash ${WORK_DIR}/scripts/eval/eval_main.sh ${DEVICE} ${MASTER_PORT} ${#GPUS[@]} ${WORK_DIR} ${CKPT_PATH} uinst/11_ ${BATCH_SIZE} $seed 10000

Thanks, i have got it. Yeah, i try to load all datasets to observe the above warning. Thanks for your reminder. By the way, your code about label smoothing seems strange, it is inconsistent with the result obtained from nn.CrossEntropyLoss(label_smoothing).

**Comment by songmzhang** [comment]
> > Yes, we did length truncation after tokenization on both prompts and responses after loading the dataset, so this warning doesn't matter. BTW, it seems that you use all the data in 11_ of the UNIST dataset for evaluation. However, we only used the first 10,000 samples in this directory for evaluation, as shown in the script.
> > [DSKD/scripts/eval/run_eval.sh](https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28)
> > Line 28 in [57d2290](/songmzhang/DSKD/commit/57d2290ce2d448be6293d97a82c6608addf33cdb)
> > bash ${WORK_DIR}/scripts/eval/eval_main.sh ${DEVICE} ${MASTER_PORT} ${#GPUS[@]} ${WORK_DIR} ${CKPT_PATH} uinst/11_ ${BATCH_SIZE} $seed 10000
> 
> Thanks, i have got it. Yeah, i try to load all datasets to observe the above warning. Thanks for your reminder. By the way, your code about label smoothing seems strange, it is inconsistent with the result obtained from nn.CrossEntropyLoss(label_smoothing).

Yeah, our code for label smoothing follows the one in fairseq. It does not affect the performance: https://github.com/facebookresearch/fairseq/blob/ecbf110e1eb43861214b05fa001eff584954f65a/fairseq/criterions/label_smoothed_cross_entropy.py#L34C1-L52C1

**Comment by sdws259** [comment]
> > > Yes, we did length truncation after tokenization on both prompts and responses after loading the dataset, so this warning doesn't matter. BTW, it seems that you use all the data in 11_ of the UNIST dataset for evaluation. However, we only used the first 10,000 samples in this directory for evaluation, as shown in the script.
> > > [DSKD/scripts/eval/run_eval.sh](https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28)
> > > Line 28 in [57d2290](/songmzhang/DSKD/commit/57d2290ce2d448be6293d97a82c6608addf33cdb)
> > > bash ${WORK_DIR}/scripts/eval/eval_main.sh ${DEVICE} ${MASTER_PORT} ${#GPUS[@]} ${WORK_DIR} ${CKPT_PATH} uinst/11_ ${BATCH_SIZE} $seed 10000
> > 
> > 
> > Thanks, i have got it. Yeah, i try to load all datasets to observe the above warning. Thanks for your reminder. By the way, your code about label smoothing seems strange, it is inconsistent with the result obtained from nn.CrossEntropyLoss(label_smoothing).
> 
> Yeah, our code for label smoothing follows the one in fairseq. It does not affect the performance: https://github.com/facebookresearch/fairseq/blob/ecbf110e1eb43861214b05fa001eff584954f65a/fairseq/criterions/label_smoothed_cross_entropy.py#L34C1-L52C1

Ok! You are one of the most responsible open-source repository authors I have seen. I have clicked "star" to show my respect.

**Comment by songmzhang** [comment]
> > > > Yes, we did length truncation after tokenization on both prompts and responses after loading the dataset, so this warning doesn't matter. BTW, it seems that you use all the data in 11_ of the UNIST dataset for evaluation. However, we only used the first 10,000 samples in this directory for evaluation, as shown in the script.
> > > > [DSKD/scripts/eval/run_eval.sh](https://github.com/songmzhang/DSKD/blob/57d2290ce2d448be6293d97a82c6608addf33cdb/scripts/eval/run_eval.sh#L28)
> > > > Line 28 in [57d2290](/songmzhang/DSKD/commit/57d2290ce2d448be6293d97a82c6608addf33cdb)
> > > > bash ${WORK_DIR}/scripts/eval/eval_main.sh ${DEVICE} ${MASTER_PORT} ${#GPUS[@]} ${WORK_DIR} ${CKPT_PATH} uinst/11_ ${BATCH_SIZE} $seed 10000
> > > 
> > > 
> > > Thanks, i have got it. Yeah, i try to load all datasets to observe the above warning. Thanks for your reminder. By the way, your code about label smoothing seems strange, it is inconsistent with the result obtained from nn.CrossEntropyLoss(label_smoothing).
> > 
> > 
> > Yeah, our code for label smoothing follows the one in fairseq. It does not affect the performance: https://github.com/facebookresearch/fairseq/blob/ecbf110e1eb43861214b05fa001eff584954f65a/fairseq/criterions/label_smoothed_cross_entropy.py#L34C1-L52C1
> 
> Ok! You are one of the most responsible open-source repository authors I have seen. I have clicked "star" to show my respect.

Really appreciate your affirmation! Please feel free to leave issues if you have any questions~

**Comment by sdws259** [comment]
Dear authors,
   I'm sorry for disturbing you again. I have met the following error when i use the script "sft_teacher_llama2.sh" to fine tune the llama2-7b model:

![Image](https://github.com/user-attachments/assets/049c3136-aae0-4689-82b8-a532038d8e39)

The error happens when evaluate the model on the dev set. However, the sft is normal when i fine tune the gpt2-base or gpt2-xl.

**Comment by sdws259** [comment]
> Dear authors, I'm sorry for disturbing you again. I have met the following error when i use the script "sft_teacher_llama2.sh" to fine tune the llama2-7b model:
> 
> ![Image](https://github.com/user-attachments/assets/049c3136-aae0-4689-82b8-a532038d8e39)
> 
> The error happens when evaluate the model on the dev set. However, the sft is normal when i fine tune the gpt2-base or gpt2-xl.

have fixed it.
